@xen-orchestra/web-core/lib/components/button/VtsButton.vue

```diff
@@ -0,0 +1,473 @@
+<!-- v4 -->
+<template>
+  <button :class="classNames" :disabled="busy || isDisabled" class="vts-button" type="button">
+    <VtsIcon :busy :icon="leftIcon" accent="current" class="icon" fixed-width />
+    <slot />
+    <VtsIcon :icon="rightIcon" accent="current" class="icon" fixed-width />
+  </button>
+</template>
+
+<script lang="ts" setup>
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import { useContext } from '@core/composables/context.composable'
+import { DisabledContext } from '@core/context'
+import type { ButtonLevel, ButtonSize } from '@core/types/button.type'
+import type { Color } from '@core/types/color.type'
+import { toVariants } from '@core/utils/to-variants.util'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = withDefaults(
+  defineProps<{
+    level: ButtonLevel
+    color: Color
+    size: ButtonSize
+    busy?: boolean
+    disabled?: boolean
+    leftIcon?: IconDefinition
+    rightIcon?: IconDefinition
+  }>(),
+  {
+    disabled: undefined,
+  }
+)
```

Comment:
According to the latest changes in the design system, I think the props’ names should be updated to `variant` for `level` and `accent` for `color` (remember to rename the props where the component is used):
```suggestion
type ButtonVariant = 'primary' | 'secondary' | 'tertiary'
type ButtonAccent = 'brand' | 'info' | 'success' | 'warning' | 'danger'

const props = withDefaults(
  defineProps<{
    variant: ButtonVariant
    accent: ButtonAccent
    size: ButtonSize
    busy?: boolean
    disabled?: boolean
    leftIcon?: IconDefinition
    rightIcon?: IconDefinition
  }>(),
  {
    disabled: undefined,
  }
)
```
I've also refactored the types, we need to add documentation about this. The types are now defined in the component (they can be exported if needed), which avoids creating a `button.type.ts` in the `/types` directory.
So the types in `button.type.ts` become useless.